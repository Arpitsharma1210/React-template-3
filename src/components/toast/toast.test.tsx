import React from 'react';
import { render, screen } from '@testing-library/react';
import Toast from './'; 
import '@testing-library/jest-dom';

describe('Toast Component', () => {
  it('renders without crashing', () => {
    render(<Toast />);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  it('renders text in ToastInfoText when text prop is provided', () => {
    render(<Toast text="Success Message" />);
    expect(screen.getByText("Success Message")).toBeInTheDocument();
  });

  it('renders subText in ToastInfoSubText when subText prop is provided', () => {
    render(<Toast subText="Additional Information" />);
    expect(screen.getByText("Additional Information")).toBeInTheDocument();
  });

  it('renders both text and subText when both props are provided', () => {
    render(<Toast text="Success Message" subText="Additional Information" />);
    expect(screen.getByText("Success Message")).toBeInTheDocument();
    expect(screen.getByText("Additional Information")).toBeInTheDocument();
  });

  it('does not crash when both text and subText are not provided', () => {
    render(<Toast />);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  it('matches snapshot when given props', () => {
    const { asFragment } = render(
      <Toast
        text="Success Message"
        subText="Additional Information"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
