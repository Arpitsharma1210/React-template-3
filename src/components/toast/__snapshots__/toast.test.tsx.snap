// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Component matches snapshot when given props 1`] = `
<DocumentFragment>
  <div
    class="sc-bcHOTY fWFYiZ"
  >
    <div
      class="sc-gsMHZj cIKVbv"
    >
      <img
        class="sc-dkHyXG kumrmy"
        src="/assets/images/success.png"
      />
    </div>
    <div
      class="sc-hLljAB jdAZhm"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 sc-eDgayY nobhg css-ahj2mt-MuiTypography-root"
      >
        Success Message
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 sc-jTkTEj jbjgy css-ahj2mt-MuiTypography-root"
      >
        Additional Information
      </p>
    </div>
  </div>
</DocumentFragment>
`;
